addons/l10n_pe_website_sale/controllers/main.py

```diff
@@ -0,0 +1,92 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+from odoo import http
+from odoo.http import request
+
+from odoo.addons.website_sale.controllers.main import WebsiteSale
+
+
+class L10nPEWebsiteSale(WebsiteSale):
+
+    def _get_mandatory_fields_billing(self, country_id=False):
+        """Extend mandatory fields to add new identification, responsibility
+        city_id and district fields when company is Peru"""
+        res = super()._get_mandatory_fields_billing(country_id)
+        company_country_id = request.website.company_id.country_id.id
+        pe_country_id = request.env.ref("base.pe").id
+        values_for_localization = ["city_id", "l10n_pe_district", "l10n_latam_identification_type_id", "vat"]
+        if country_id == pe_country_id and company_country_id == pe_country_id:
+            res += values_for_localization
+            res.remove("city")
+        elif company_country_id != pe_country_id and "city" not in res:
+            res += ["city"]
+            new_res = [value for value in res if value not in values_for_localization]
+            res = new_res
+        return res
+
+    def _get_mandatory_fields_shipping(self, country_id=False):
+        """Extend mandatory fields to add city_id and district fields when the selected country is Peru"""
+        res = super()._get_mandatory_fields_shipping(country_id)
+        company_country_id = request.website.company_id.country_id.id
+        pe_country_id = request.env.ref("base.pe").id
+        values_for_localization = ["city_id", "l10n_pe_district"]
+        if country_id == pe_country_id and company_country_id == pe_country_id:
+            res += values_for_localization
+            res.remove("city")
+        elif company_country_id != pe_country_id and "city" not in res:
+            res += ["city"]
+            new_res = [value for value in res if value not in values_for_localization]
```

Comment:
Because it uses a different field for cities and districts that work differently than the native behavior.

Reply:
In the "if" we remove it because the country is Peru and we don't need the default value of city which is taken from a text input from our Billing form, instead we need to take it from a select input called city_id. In the "elif" we add it back when the country is not Peru so we can take that value from the original text input just as it works natively.